### PR TITLE
[#362][#522][#523] allow '=' and ';' in PAM passwords

### PR DESCRIPTION
**Pull request by d-w-moore**

### CODE REVIEW

File: irods/test/PRC_issue_362.bats [hunk 1]
```diff
@@ -12,25 +13,43 @@ setup() {
...
+}
 
-@test "test_with_atsymbol"	{ prc_test; }
-@test "test_with_semicolon"	{ prc_test; }
-@test "test_with_equals"	{ prc_test; }
-@test "test_with_ampersand"	{ prc_test; }
+@test "test_with_atsymbol" { prc_test; }
+@test "test_with_semicolon" { prc_test; }
+@test "test_with_equals" { prc_test; }
+@test "test_with_ampersand" { prc_test; }
```
Comment (korydraughn) [inline irods/test/PRC_issue_362.bats:55]: Are the tests actually different? If yes, how/where is `prc_test` being changed?

File: irods/connection.py [hunk 2]
```diff
@@ -490,9 +488,10 @@ def _login_pam(self):
             if getattr(self,'DISALLOWING_PAM_PLAINTEXT',True):
                 raise PlainTextPAMPasswordError
 
-        Pam_Long_Tokens = (ALLOW_PAM_LONG_TOKENS and (len(ctx) >= MAX_NAME_LEN))
+        use_dedicated_pam_api = len(ctx) >= MAX_NAME_LEN or \
+                                {';','='}.intersection(set(new_pam_password))
```
[the remark this answers is not in the document]
Comment (d-w-moore) [reply irods/connection.py:492]: Right

File: irods/test/PRC_issue_362.bats (hunk 1 above)
Comment (d-w-moore) [reply irods/test/PRC_issue_362.bats:55]: Its a hack actually where we change a global variable based on the name of the test, $BATS_TEST_NAME.

Yeah, ick.  I can change that easily

Comment (d-w-moore) [reply irods/test/PRC_issue_362.bats]: Done

File: irods/connection.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (d-w-moore) [reply irods/connection.py:494]: Done

**Review by alanking — COMMENTED** [review]
Seems good to me. Tests are passing?

**Comment by d-w-moore** [comment]
> Seems good to me. Tests are passing?

I think we'll be testing a few pull requests at once; see [this comment](https://github.com/irods/python-irodsclient/pull/530#issuecomment-2047561615)

**Comment by d-w-moore** [comment]
Tested and passing.

**Comment by korydraughn** [comment]
Squash to taste if ready, no pounds.

**Review by alanking — APPROVED** [review]

**Pull request merged**